manifest.md

```diff
@@ -191,6 +191,10 @@ Unlike the [Manifest List](#manifest-list), which contains information about a s
 
         This REQUIRED property is the digest of the content, as defined by the [Descriptor](descriptor.md) digest format.
 
+    - **`urls`** *array*
+
+       For an ordinary layer, this is empty, and the layer contents can be retrieved directly from the registry. For a layer with mediatype of application/vnd.docker.image.rootfs.foreign.diff.tar.gzip, this contains a non-empty list of URLs from which this object can be downloaded.
```

Comment:
On Wed, Jul 06, 2016 at 10:41:39AM -0700, W. Trevor King wrote:
> > +       For an ordinary layer, this is empty, and the layer contents can be retrieved directly from the registry. For a layer with mediatype of application/vnd.docker.image.rootfs.foreign.diff.tar.gzip, this contains a non-empty list of URLs from which this object can be downloaded.
>
> The `application/vnd.docker.image.rootfs.foreign.diff.tar.gzip`
> media type is not defined in this spec…

Now that I have a clearer idea of what this media type is for [1], I
think a better approach is to make ‘foreign’ (or some such) an
explicit property of wherever the mutable mirror data is stored [2].
Then the publisher is saying “you need these blobs to build my image”,
but not telling you how to get them.  And the distributor (managing
the mutable mirror data) is saying “you can get that blob from the
usual place and also these mirrors” or “you can't get that blob from
the usual place and have to get it from one of these mirrors”.

Clients who choose not to hit the mutable mirror engine and go to the
usual place for foreign blobs will get the usual key-not-found error
(and potentially a hint pointing at a particular mutable mirror
engine), but that's an acceptable workflow to my eyes.

[1]: https://github.com/opencontainers/image-spec/pull/169#issuecomment-230906535
[2]: https://github.com/opencontainers/image-spec/pull/169#issuecomment-230849934